File: ValidCPF.py
#Valide o CPF

import re

def validCPF (cpf):

	soma = 0
	digito = 0

	if len(cpf) != 11: return False

	#primeiro digito
	j = 0		
	for i in range (10, 1, -1):
		soma = soma + (int(cpf[j])*i)
		j += 1

	digito = (soma % 11)

	if digito == 10 or digito == 11: digito = 0

	digito = 11 - digito

	if digito != int(cpf[9]): return False

	#segundo digito
	soma = 0
	j = 0		
	for i in range (11, 1, -1):
		soma = soma + (int(cpf[j])*i)
		j += 1

	digito = (soma % 11)

	if digito == 10 or digito == 11: digito = 0

	digito = 11 - digito
	
	if digito != int(cpf[10]): return False

	return True

cpf = input("Digite o CPF: ")

"""
for char in ".-":
	cpf = cpf.replace(char, "")
"""

cpf = re.sub('\D', "", cpf)

#print(cpf)

if validCPF(cpf):
	print("CPF Válido :D")
else:
	print("CPF Inválido :(")

File: ValidCPF.js
//Valide o CPF

function validCPF (cpf){

	let soma = 0, digito = 0, i, j
	
	if(cpf.length != 11) return false
		
		//primeiro digito
		j = 0		
		for (i = 10; i>1; i--){
			soma = soma + (parseInt(cpf[j])*i)
			j++
		}

		digito = (soma % 11)

		if(digito == 10 || digito == 11) digito = 0

		digito = 11 - digito
		
		if(digito != parseInt(cpf[9])) return false

		//segundo digito
		soma = 0
		j = 0		
		for (i = 11; i>1; i--){
			soma = soma + (parseInt(cpf[j])*i)
			j++
		}
		digito = (soma % 11)

		if(digito == 10 || digito == 11) digito = 0

		digito = 11 - digito
		
		if(digito != parseInt(cpf[10])) return false

	return true
}

//const cpfString = "546.471.429-49"
const cpfString = prompt("Digite seu CPF ")

const cpf = cpfString.replace(/\D+/g, '')

//console.log(cpf);

if(validCPF(cpf))
	console.log("CPF Válido :D")
else
	console.log("CPF Inválido :(")